pkg/api/util/annotations.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const (
+	metadataAnnotationPattern string = "^metadata.annotations\\['((.)+)'\\]$"
+)
+
+type AnnotationFieldPathParser struct {
+	annotationRegexp *regexp.Regexp
+}
+
+func NewAnnotationFieldPathParser() (*AnnotationFieldPathParser, error) {
+	annotationRegexp, err := regexp.Compile(metadataAnnotationPattern)
```

Comment:
Maybe use `MustCompile()` instead so that the caller doesn't need to handle errors.